apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -0,0 +1,159 @@
+import {AsyncPipe} from '@angular/common';
+import {
+  Component,
+  inject,
+  Input,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+  ViewEncapsulation
+} from '@angular/core';
+import {FormBuilder, FormGroup, FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {FileSystemTree} from '@webcontainer/api';
+import {MonacoEditorModule} from 'ngx-monaco-editor-v2';
+import {combineLatest, debounceTime, distinctUntilChanged, filter, map, skip, startWith, Subscription} from 'rxjs';
+import {WebContainerService} from '../../../services';
+import {CodeEditorControlComponent} from '../code-editor-control';
+import {NgxMonacoTreeComponent} from 'ngx-monaco-tree';
+
+/** ngx-monaco-editor options language - determined based on file extension */
+const editorOptionsLanguage: Record<string, string> = {
+  html: 'xml',
+  json: 'json',
+  ts: 'typescript',
+  js: 'javascript'
+};
+
+/** Mode of Code Editor */
+export type EditorMode = 'readonly' | 'interactive';
+
+/** Project properties for the current exercise */
+export interface TrainingProject {
+  /** Starting file to be displayed in the project */
+  startingFile: string;
+  /** Files in the project */
+  files: FileSystemTree;
+  /** Commands to run in the project */
+  commands: string[];
+  /** Current working directory in project */
+  cwd: string;
+}
+
+@Component({
+  selector: 'code-editor-view',
+  standalone: true,
+  imports: [
+    AsyncPipe,
+    FormsModule,
+    MonacoEditorModule,
+    ReactiveFormsModule,
+    NgxMonacoTreeComponent,
+    CodeEditorControlComponent
+  ],
+  encapsulation: ViewEncapsulation.None,
+  templateUrl: './code-editor-view.component.html',
+  styleUrl: './code-editor-view.component.scss'
+})
+export class CodeEditorViewComponent implements OnDestroy, OnChanges {
+
+  @Input() public editorMode: EditorMode = 'readonly';
+  @Input() public project?: TrainingProject;
+
+  /** Variable to display "Log tree" button during debug mode */
```

Comment:
is that really helpful ?

Reply:
For now, this is our only way to export an exercise.